reporter/src/main/kotlin/reporters/SpdxDocumentReporter.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import java.io.File
+
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.SpdxDocumentModelMapper
+import org.ossreviewtoolkit.spdx.SpdxModelSerializer
+
+/**
+ * Creates YAML and JSON SPDX documents. The option keys [CREATION_INFO_COMMENT], [DOCUMENT_COMMENT] and [DOCUMENT_NAME]
+ * can be provided to [generateReport] to inject the corresponding values as meta-data into the produced [SpdxDocument].
+ * The option key [OUTPUT_FILE_FORMATS] specifies a comma separated list of [FileFormat]s, whereas [FileFormat.values]
+ * is used by default.
+ */
+class SpdxDocumentReporter : Reporter {
+    companion object {
+        const val CREATION_INFO_COMMENT = "creationInfo.comment"
+        const val DOCUMENT_COMMENT = "document.comment"
+        const val DOCUMENT_NAME = "document.name"
+        const val OUTPUT_FILE_FORMATS = "output.file.formats"
+
+        private const val DOCUMENT_NAME_DEFAULT_VALUE = "Unnamed document"
+    }
+
+    enum class FileFormat(val reportFilename: String) {
+        JSON("document.spdx.json"),
+        YAML("document.spdx.yml")
+    }
+
+    override val reporterName = "SpdxDocument"
```

Comment:
I see the name for this reporter is `SpdxDocument` why not use just `Spdx` .. less typing for our user ;-)

Reply:
I preferred `SpdxDocument` as it is called this way in the specification and in our code, as the reporter outputs `SpdxDocuments` and for consistency with `Antenna*Reporter`.